src/Directions.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+
+import { MapContext } from './Map'
+
+import { createCoordinate, NumberTuple } from './utils'
+
+type annotationOptions = {
+  origin?: mapkit.MarkerAnnotationConstructorOptions
+  destination?: mapkit.MarkerAnnotationConstructorOptions
+}
+
+type DirectionsProps = {
+  origin: NumberTuple
+  destination: NumberTuple
+  transportType?: 'Automobile' | 'Walking'
+  style?: mapkit.StyleConstructorOptions
+  options?: mapkit.DirectionsConstructorOptions
+  showAnnotations?: boolean
+  annotationOptions?: annotationOptions
+  routeDetails?: (arg0: {
+    steps: mapkit.RouteStep[]
+    distance: number
+    expectedTravelTime: number
+  }) => void
+}
+
+export const Directions: React.FC<DirectionsProps> = ({
+  origin,
+  destination,
+  transportType = 'Automobile',
+  style,
+  showAnnotations = true,
+  annotationOptions = {
+    origin: { color: 'red' },
+    destination: { color: 'green' },
+  },
+  options,
+  routeDetails = () => null,
+}) => {
+  const { mapkit, map } = React.useContext(MapContext)
+
+  React.useEffect(() => {
+    if (mapkit && map) {
+      const directions = new mapkit.Directions(options)
+      const originCoordinates = createCoordinate(
+        origin[0] as number,
+        origin[1] as number,
+      )
+      const destinationCoordinates = createCoordinate(
+        destination[0] as number,
+        destination[1] as number,
+      )
+
+      if (showAnnotations) {
+        const originAnnotation = new mapkit.MarkerAnnotation(
+          originCoordinates,
+          {
+            ...annotationOptions.origin,
+            draggable: false,
+          },
+        )
+
+        const destinationAnnotation = new mapkit.MarkerAnnotation(
+          destinationCoordinates,
+          {
+            ...annotationOptions.destination,
+            draggable: false,
+          },
+        )
+
+        map.addAnnotations([originAnnotation, destinationAnnotation])
+      }
+
+      directions.route(
+        {
+          origin: originCoordinates,
+          destination: destinationCoordinates,
+          transportType:
```

Comment:
Ah, I see. I wonder if we could future proof this by using `mapkit.Directions.Transport[transportType]` and making sure the annotation on the prop uses the values provided by mapkit?

Reply:
By modifying the transportType to be of type `mapkit.Directions.Transport` with the prop being the same. This does require a user to ender into the prop `mapkit.Directions.Transport.Automobile` or `mapkit.Directions.Transport.Walking`